### ISSUE TITLE
SOLPS grid integrator

### ISSUE THREAD

**Issue by mattngc** [cherab/solps#3]
* Current version of the emission models defaults to the NumericalIntegrator(step=x cm). The class essentially performs numerical integration using the given step size.
* A potentially much faster and more powerful solution is to use the kd-tree to find all intersecting cells and calculate their traversal lengths analytically. This should lead to a much faster and more accurate calculation method.

**Comment by jrh-ccfe** [comment]
Out of curiosity, how much work is involved in switching to this improved integrator?  For production runs it could potentially save great deal of time, and removes the requirement for the user to specify a step length.

**Comment by mattngc** [comment]
Yes, it potentially adds ridiculous speed ups. I wouldn't be surprised if you could get almost a factor x10 on SOLPS based runs. So much time is wasted on the sub-optimal integration scheme for these problems.

Its probably one of the most technical features of Raysect though. Definitely not for non-experts.  Sad thing is, I had a working draft on one of my private branches a few years ago, but I'm not sure what happened to it. Somehow it never got merged. I'll have to dig around and see if I can find the draft code again. But if it was on my CCFE PC... I'd have to start again. I think I remember how it worked though.

To answer your original question, probably 2 weeks for someone who understands kd-trees and the framework to a deep enough level.
